ankihub/main/block_exam_subdecks.py

```diff
@@ -144,27 +145,136 @@ def check_block_exam_subdeck_due_dates() -> List[BlockExamSubdeckConfig]:
     return expired_subdecks
 
 
-def trigger_due_date_dialog(expired_subdecks: List[BlockExamSubdeckConfig]) -> None:
-    """Trigger the Due Date Dialog for expired subdecks.
+def move_subdeck_to_main_deck(subdeck_config: BlockExamSubdeckConfig) -> bool:
+    """Move all notes from a subdeck back to the main deck and delete the subdeck.
+
+    Args:
+        subdeck_config: Configuration of the subdeck to move
+
+    Returns:
+        bool: True if successful, False otherwise
+    """
+    try:
+        ankihub_deck_id = uuid.UUID(subdeck_config.ankihub_deck_id)
+        deck_config = config.deck_config(ankihub_deck_id)
+        if not deck_config:
+            LOGGER.error("Deck config not found for moving subdeck", ankihub_deck_id=str(ankihub_deck_id))
+            return False
+
+        subdeck_id = DeckId(int(subdeck_config.subdeck_id))
+        subdeck = aqt.mw.col.decks.get(subdeck_id, default=False)
+        if not subdeck:
+            LOGGER.warning("Subdeck not found, removing config", subdeck_id=subdeck_config.subdeck_id)
+            remove_block_exam_subdeck_config(subdeck_config)
+            return True
+
+        parent_deck_id = deck_config.anki_id
+
+        note_ids = aqt.mw.col.find_notes(f"did:{subdeck_id}")
```

Comment:
:wrench: This will miss notes which are in filtered decks and have their `odid` (original deck id) set to the subdeck.

We can use this instead:
```suggestion
        note_ids = aqt.mw.col.db.list("SELECT DISTINCT nid FROM cards WHERE did=? OR odid=?", subdeck_id, subdeck_id)
```